cloud-controller-manager/do/loadbalancers.go

```diff
@@ -245,6 +256,15 @@ func (l *loadBalancers) GetLoadBalancerName(_ context.Context, clusterName strin
 }
 
 func getDefaultLoadBalancerName(service *v1.Service) string {
+	name := service.Annotations[annoDOLoadBalancerName]
```

Comment:
It might be possible to resolve the name uniqueness problem here, by introducing another annotation we would use to track the _slugified_ name with a few random chars added at the end. My understanding of the problem here is that there is no good mechanism of surfacing errors that come from creation / reconciliation of a LoadBalancer service. I might be mistaken here, of course, but this seems plausible.

If that is indeed the case, we can introduce another annotation `kubernetes.digitalocean.com/load-balancer-name-id` that would be a slugified and unique name, generated by the cloud controller from the  user-provided `kubernetes.digitalocean.com/load-balancer-name`.

Having `load-balancer-name-id` would resolve the problem of uniqueness while making it possible to specify a user-friendly load balancer name. It would make this feature a little more robust too. It adds some amount of complexity, but not too much, and it might be a good balance. WDYT @timoreimann?

Reply:
That's an interesting idea. In general, I'm hesitant to introduce more annotations than necessary: it has become quite challenging to fully comprehend the number of annotations (and their interactions) we support, which also matches with user feedback we have received.

I think we may not need that slugified name annotation anymore though -- please see my general review comment.